src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQuerySnapshotPdao.java

```diff
@@ -1092,16 +1072,58 @@ private void storeRowIdsForRelatedTable(
     String liveViewSqlTo =
         BigQueryDatasetPdao.renderDatasetLiveViewSql(
             datasetProjectId, datasetBqDatasetName, toTable, null, filterBefore);
+
+    ST fromTableTableSelect;
+    ST toTableTableSelect;
+    String fromCol;
+    String toCol;
+    if (relationship.getFromColumnIsArray() && relationship.getToColumnIsArray()) {
+      fromTableTableSelect =
+          new ST(tableSelectArray)
+              .add("toOrFrom", "FROM")
+              .add("field", relationship.getFromColumnName());
+      toTableTableSelect =
+          new ST(tableSelectArray)
+              .add("toOrFrom", "TO")
+              .add("field", relationship.getToColumnName());
+      fromCol = "FLAT_FROM";
+      toCol = "FLAT_TO";
+    } else if (relationship.getFromColumnIsArray()) {
+      fromTableTableSelect =
+          new ST(tableSelectArray)
+              .add("toOrFrom", "FROM")
+              .add("field", relationship.getFromColumnName());
+      toTableTableSelect = new ST(tableSelectNonArray);
+      fromCol = "FLAT_FROM";
+      toCol = relationship.getToColumnName();
+    } else if (relationship.getToColumnIsArray()) {
+      fromTableTableSelect = new ST(tableSelectNonArray);
+      toTableTableSelect =
+          new ST(tableSelectArray)
+              .add("toOrFrom", "TO")
+              .add("field", relationship.getToColumnName());
+      fromCol = relationship.getFromColumnName();
+      toCol = "FLAT_TO";
+    } else {
+      fromTableTableSelect = new ST(tableSelectNonArray);
+      toTableTableSelect = new ST(tableSelectNonArray);
+      fromCol = relationship.getFromColumnName();
+      toCol = relationship.getToColumnName();
+    }
```

Comment:
Is there a reason that you've handled these assignments in this combinatorial way?

I think that handling all the possible from-to combinations in assignment overly complicates things and repeats code.

What about ternary assignments?
```
ST fromTableTableSelect =
  (relationship.getFromColumnIsArray()) ?
  new ST(tableSelectArray).add("toOrFrom", "FROM").add("field", relationship.getFromColumnName()) : 
  new ST(tableSelectNonArray);
String fromCol =
  (relationship.getFromColumnIsArray()) ?
  "FLAT_FROM" : 
  relationship.getFromColumnName();
```

Or handling all the to-or-from assignments in their own block?
```
if (relationship.getFromColumnIsArray()) {
  fromTableTableSelect = …
  fromCol = …
} else {
  fromTableTableSelect = …
  fromCol = …
}
```